sql/src/main/java/io/crate/expression/reference/file/LineContext.java

```diff
@@ -66,8 +58,27 @@ public Object get(ColumnIdent columnIdent) {
         return val;
     }
 
+    public Map<String, Object> sourceAsMap() {
+        if (parsedSource == null) {
+            try {
+                parsedSource = XContentHelper.convertToMap(new BytesArray(rawSource), false, XContentType.JSON).v2();
+            } catch (NullPointerException e) {
+                return null;
+            }
+        }
+        return parsedSource;
+    }
+
     public void rawSource(byte[] bytes) {
         this.rawSource = bytes;
         this.parsedSource = null;
     }
+
+    public void rawSourceFromCSV(byte[] header, byte[] line) throws IOException {
+        CSVLineParser csvParser = new CSVLineParser();
+        String convertedCsvToJsonString = csvParser.parse(header, line);
+
+        this.rawSource = convertedCsvToJsonString.getBytes(StandardCharsets.UTF_8);
```

Comment:
If I'm not mistaken you parse CSV to JSON here which gets parsed again later on. That looks like we are doing the work twice.

If we have the dedicated LineContexts for csv/json, we could directly build the source map here return by sourceAsMap. What do you think?